server/src/main/java/org/elasticsearch/cluster/coordination/NodeJoinExecutor.java

```diff
@@ -155,15 +157,6 @@ public ClusterState execute(BatchExecutionContext<JoinTask> batchExecutionContex
                 final DiscoveryNode node = nodeJoinTask.node();
                 if (currentNodes.nodeExistsWithSameRoles(node)) {
                     logger.debug("received a join request for an existing node [{}]", node);
-
-                    // update the node's feature set if it has one
-                    // this can happen if the master has just moved from a pre-features version to a post-features version
-                    assert Version.V_8_12_0.onOrBefore(Version.CURRENT) : "This can be removed once 8.12.0 is no longer a valid version";
```

Comment:
Taking the opportunity to remove this (and the associated test) to simplify managing the node feature collections